pc-air/pc_variant_correlation.wdl

```diff
@@ -0,0 +1,325 @@
+version 1.0
+
+task pca_corr_vars {
+    input {
+        File gds_file
+        String chromosome
+
+        File? variant_include_file
+        String? out_prefix
+        Int? n_corr_vars
+        
+        # runtime attributes
+        Int addldisk = 5
+        Int cpu = 2
+        Int memory = 4
+        Int preempt = 3
+    }
+
+    # Estimate disk size required
+    Int gds_size        = ceil(size(gds_file, "GB"))
+    Int final_disk_dize = gds_size + addldisk
+
+    # Are optional files defined?
+    Boolean defVariantInclude = defined(variant_include_file)
+    
+
+    command <<<
+        set -eux -o pipefail
+
+        echo "Generating config file"
+        # noquotes = argument has no quotes in the config file to match CWL config EXACTLY
+        python << CODE
+        import os
+
+        f = open("pca_corr_vars.config", "a")
+
+        if "~{out_prefix}" != "":
+            f.write('out_file "~{out_prefix}_pca_corr_vars_chr .RData"\n')
+        else:
+            f.write('out_file "pca_corr_vars_chr .RData"\n')
+        
+        if "~{defVariantInclude}" == "true":
+            f.write('variant_include_file "~{variant_include_file}"\n') 
+
+        f.write('gds_file "~{gds_file}"\n')
+
+        f.write('segment_file "/usr/local/analysis_pipeline/segments_hg38.txt"\n')
+
+        f.close()
+
+        CODE
+
+        echo "Calling R script pca_corr_vars.R"
+        Rscript /usr/local/analysis_pipeline/R/pca_corr_vars.R pca_corr_vars.config --chromosome ~{chromosome}
```

Comment:
The CWL notes that 23=X and 24=Y but I think earlier chromosome number splitting would still have these as X and Y. I might have lost the plot here, when do these get transformed?

Reply:
They don't. The CWL just ends up passing a string of X/Y. I confirmed by running in SB.